pkg/operation/botanist/nodelocaldns.go

```diff
@@ -50,14 +50,15 @@ func (b *Botanist) DefaultNodeLocalDNS() (nodelocaldns.Interface, error) {
 		b.SeedClientSet.Client(),
 		b.Shoot.SeedNamespace,
 		nodelocaldns.Values{
-			Image:                 image.String(),
-			VPAEnabled:            b.Shoot.WantsVerticalPodAutoscaler,
-			ForceTcpToClusterDNS:  v1beta1helper.IsTCPEnforcedForNodeLocalDNSToClusterDNS(b.Shoot.GetInfo().Spec.SystemComponents, b.Shoot.GetInfo().Annotations),
-			ForceTcpToUpstreamDNS: v1beta1helper.IsTCPEnforcedForNodeLocalDNSToUpstreamDNS(b.Shoot.GetInfo().Spec.SystemComponents, b.Shoot.GetInfo().Annotations),
-			ClusterDNS:            clusterDNS,
-			DNSServer:             dnsServer,
-			PSPDisabled:           b.Shoot.PSPDisabled,
-			KubernetesVersion:     b.Shoot.KubernetesVersion,
+			Image:                       image.String(),
+			VPAEnabled:                  b.Shoot.WantsVerticalPodAutoscaler,
+			ForceTcpToClusterDNS:        v1beta1helper.IsTCPEnforcedForNodeLocalDNSToClusterDNS(b.Shoot.GetInfo().Spec.SystemComponents, b.Shoot.GetInfo().Annotations),
+			ForceTcpToUpstreamDNS:       v1beta1helper.IsTCPEnforcedForNodeLocalDNSToUpstreamDNS(b.Shoot.GetInfo().Spec.SystemComponents, b.Shoot.GetInfo().Annotations),
+			DisableForwardToUpstreamDNS: v1beta1helper.IsDisabledNodeLocalDNSForwardToUpstreamDNS(b.Shoot.GetInfo().Spec.SystemComponents),
```

Comment:
Instead of adding/duplicating each field of the `gardencorev1beta1.NodeLocalDNS` type here in the `Values`, shouldn't we simply pass the entire `.spec.systemComponents.nodeLocalDNS` to this component?